File: python/71. Simplify Path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        for temp in path.split("/"):
            if temp == "..":
                if stack:
                    stack.pop()
            elif temp != "." and temp != "":
                stack.append(temp)
        return "/" + ("/".join(stack) if stack else "")

File: cpp/array/71. Simplify Path.cpp
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
    string simplifyPath(string path) {
        vector<string> stack;
        int n = path.size();

        for (int i = 0, begin = 0; i <= n; i++) {
            if (i == n || path[i] == '/') {
                string temp = path.substr(begin, i - begin);
                if (temp == "..") {
                    if (!stack.empty())
                        stack.pop_back();
                }
                else if (temp != "." && !temp.empty())
                    stack.push_back(temp);
                begin = i+1;
            }
        }

        string res; 
        for (const auto &s : stack) res += "/" + s;
        return res.empty() ? "/" : res; 
    }
};